### PR TITLE
Fix recording on the Sony Z5

### PR DESCRIPTION
**Pull request by emmaguy**
Tested on S5 mini, S6, S7 and Z5, acer AV0L0 and 5X

### CODE REVIEW

File: library/src/main/camera2/com/google/android/cameraview/Camera2.java [hunk 1]
```diff
@@ -117,12 +118,14 @@ public void onClosed(@NonNull CameraDevice camera) {
 
         @Override
         public void onDisconnected(@NonNull CameraDevice camera) {
+            camera.close();
```
Comment (emmaguy) [inline library/src/main/camera2/com/google/android/cameraview/Camera2.java:121]: Clean up the camera after it's disconnected or errored

File: library/src/main/camera2/com/google/android/cameraview/Camera2.java [hunk 2]
```diff
@@ -139,17 +142,22 @@ public void onConfigured(@NonNull CameraCaptureSession session) {
...
-                mRecording = true;
-            }
-            try {
-                mCaptureSession.setRepeatingRequest(mPreviewRequestBuilder.build(),
-                        mCaptureCallback, null);
-            } catch (CameraAccessException | IllegalStateException e) {
-                Log.e(TAG, "Failed to start camera preview.", e);
-            }
+
+            new Handler(Looper.getMainLooper()).post(new Runnable() {
```
Comment (emmaguy) [inline library/src/main/camera2/com/google/android/cameraview/Camera2.java:146]: We run the callback on a background thread now - but we still need to start mediarecorder on the main thread

File: library/src/main/camera2/com/google/android/cameraview/Camera2.java [hunk 3]
```diff
@@ -625,10 +650,10 @@ private void startCaptureSession() {
             if (mStartVideoRecording) {
                 mPreviewRequestBuilder.addTarget(mMediaRecorder.getSurface());
                 mCamera.createCaptureSession(Arrays.asList(surface, mMediaRecorder.getSurface()),
-                                             mSessionCallback, new Handler(Looper.getMainLooper()));
+                                             mSessionCallback, mBackgroundHandler);
```
Comment (emmaguy) [inline library/src/main/camera2/com/google/android/cameraview/Camera2.java:653]: capture on the background thread

**Review by ivacf — APPROVED** [review]
📸

**Pull request merged**